Symfony/CS/Fixer/PSR2/EofEndingFixer.php

```diff
@@ -21,17 +22,16 @@ class EofEndingFixer extends AbstractFixer
     /**
      * {@inheritdoc}
      */
-    public function fix(\SplFileInfo $file, $content)
+    public function fix(\SplFileInfo $file, Tokens $tokens)
     {
         // [Structure] A file must always end with a linefeed character
-
+        $content = $tokens->generateCode();
         $content = rtrim($content);
-
         if (strlen($content)) {
-            return $content."\n";
+            $content .= "\n";
         }
 
-        return $content;
+        $tokens->setCode($content);
```

Comment:
couldn't we handle it by checking that the last token of the collection ends with a EOL instead of reparsing the whole file ?